src/cli/config-flags.ts

```diff
@@ -187,26 +188,39 @@ export const LOG_LEVEL_CLI_ARGS = ['logLevel'] as const;
  */
 type ArrayValuesAsUnion<T extends ReadonlyArray<string>> = T[number];
 
-export type BooleanCLIArg = ArrayValuesAsUnion<typeof BOOLEAN_CLI_ARGS>;
-export type StringCLIArg = ArrayValuesAsUnion<typeof STRING_CLI_ARGS>;
-export type NumberCLIArg = ArrayValuesAsUnion<typeof NUMBER_CLI_ARGS>;
-export type StringNumberCLIArg = ArrayValuesAsUnion<typeof STRING_NUMBER_CLI_ARGS>;
-export type LogCLIArg = ArrayValuesAsUnion<typeof LOG_LEVEL_CLI_ARGS>;
+export type BooleanCLIFlag = ArrayValuesAsUnion<typeof BOOLEAN_CLI_FLAGS>;
+export type StringCLIFlag = ArrayValuesAsUnion<typeof STRING_CLI_FLAGS>;
+export type StringArrayCLIFlag = ArrayValuesAsUnion<typeof STRING_ARRAY_CLI_FLAGS>;
+export type NumberCLIFlag = ArrayValuesAsUnion<typeof NUMBER_CLI_FLAGS>;
+export type StringNumberCLIFlag = ArrayValuesAsUnion<typeof STRING_NUMBER_CLI_FLAGS>;
+export type LogCLIFlag = ArrayValuesAsUnion<typeof LOG_LEVEL_CLI_FLAGS>;
 
-type KnownCLIArg = BooleanCLIArg | StringCLIArg | NumberCLIArg | StringNumberCLIArg | LogCLIArg;
+export type KnownCLIFlag =
+  | BooleanCLIFlag
+  | StringCLIFlag
+  | StringArrayCLIFlag
+  | NumberCLIFlag
+  | StringNumberCLIFlag
+  | LogCLIFlag;
 
-type AliasMap = Partial<Record<KnownCLIArg, string>>;
+type AliasMap = Partial<Record<string, KnownCLIFlag>>;
 
 /**
  * For a small subset of CLI options we support a short alias e.g. `'h'` for `'help'`
  */
-export const CLI_ARG_ALIASES: AliasMap = {
-  config: 'c',
-  help: 'h',
-  port: 'p',
-  version: 'v',
+export const CLI_FLAG_ALIASES: AliasMap = {
+  c: 'config',
+  h: 'help',
+  p: 'port',
+  v: 'version',
 };
 
+/**
+ * A regular expression which can be used to match a CLI flag for one of our
+ * short aliases.
+ */
+export const CLI_FLAG_REGEX = new RegExp(`^-[chpv]{1}$`);
```

Comment:
I actually had basically exactly that originally but I ended up having to change it because it caused a tree shaking issue, I think because then `CLI_FLAG_REGEX` had a dependency on `CLI_FLAG_ALIASES` so the treeshaking check that we do for the build verification broke 😢

Reply:
which does introduce a bit of an unfortunate situation where we can't automatically keep the regex in sync with the defined aliases. If we have some precedent for adding an exception for the tree-shaking build verification thingy I'd be open to doing that